full_name.rb

```diff
@@ -0,0 +1,11 @@
+def new_array 
+  name_array = []
+  puts "What is your first name?"
+  name_array.push(gets.chomp)
+  puts "What is your middle name?"
+  name_array.push(gets.chomp)
+  puts "What is your last name?"
+  name_array.push(gets.chomp)
+  puts "Hi there #{name_array[0]} #{name_array[1]} #{name_array[2]}!"
+end
+new_array
```

Comment:
This code is good. It is important to have some sort of validation with the inputs (I would say mainly in first name and last name) because, if we leave them blank, this is the final message:
`Hi there   !`

We can add some sort of conditional. If the user doesn't enter anything either first or last name, the program returns a message saying something like "This field is required". This is not mandatory to do, however, it is a good practice always making "full proof" programs.